From 568e5d6fa3458021ec0eab5e96ba21738e4b0356 Mon Sep 17 00:00:00 2001
From: Brecht Van Lommel <brecht@blender.org>
Date: Thu, 12 Jun 2025 14:43:47 +0200
Subject: [PATCH 17/17] BLENDER: Fine tune pages considered expensive

---
 routers/common/blockexpensive.go | 32 ++++++++++++++++----------------
 1 file changed, 16 insertions(+), 16 deletions(-)

diff --git a/routers/common/blockexpensive.go b/routers/common/blockexpensive.go
index f52aa2b709..b0c3aee079 100644
--- a/routers/common/blockexpensive.go
+++ b/routers/common/blockexpensive.go
@@ -39,28 +39,28 @@ func isRoutePathExpensive(routePattern string) bool {
 	}
 
 	expensivePaths := []string{
-		// code related
+		// code related, very expensive pages
 		"/{username}/{reponame}/archive/",
 		"/{username}/{reponame}/blame/",
-		"/{username}/{reponame}/commit/",
-		"/{username}/{reponame}/commits/",
 		"/{username}/{reponame}/graph",
-		"/{username}/{reponame}/media/",
-		"/{username}/{reponame}/raw/",
-		"/{username}/{reponame}/src/",
 
-		// issue & PR related (no trailing slash)
-		"/{username}/{reponame}/issues",
-		"/{username}/{reponame}/{type:issues}",
-		"/{username}/{reponame}/pulls",
-		"/{username}/{reponame}/{type:pulls}",
-
-		// wiki
-		"/{username}/{reponame}/wiki/",
+		// activity, trailing slash removed
+		"/{username}/{reponame}/activity",
+	}
 
-		// activity
-		"/{username}/{reponame}/activity/",
+	if !(strings.HasPrefix(routePattern, "/blender/") ||
+		strings.HasPrefix(routePattern, "/studio/") ||
+		strings.HasPrefix(routePattern, "/extensions/") ||
+		strings.HasPrefix(routePattern, "/infrastructure/")) {
+		// code related, less expensive not allowed in forks
+		expensivePaths = append(expensivePaths,
+			"/{username}/{reponame}/media/",
+			"/{username}/{reponame}/commit/",
+			"/{username}/{reponame}/commits/",
+			"/{username}/{reponame}/src/",
+			"/{username}/{reponame}/raw/")
 	}
+
 	for _, path := range expensivePaths {
 		if strings.HasPrefix(routePattern, path) {
 			return true
-- 
2.49.0

