From f4ae3012908b3098011d6a9dd420d91de080bf3d Mon Sep 17 00:00:00 2001
From: Brecht Van Lommel <brecht@blender.org>
Date: Thu, 12 Jun 2025 14:43:47 +0200
Subject: [PATCH 1/2] BLENDER: Don't consider issues/pulls/wiki/activity as
 expensive pages

---
 routers/common/blockexpensive.go | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/routers/common/blockexpensive.go b/routers/common/blockexpensive.go
index ba11ff4cf9..3a1e0ba250 100644
--- a/routers/common/blockexpensive.go
+++ b/routers/common/blockexpensive.go
@@ -48,19 +48,6 @@ func isRoutePathExpensive(routePattern string) bool {
 		"/{username}/{reponame}/media/",
 		"/{username}/{reponame}/raw/",
 		"/{username}/{reponame}/src/",
-
-		// issue & PR related (no trailing slash)
-		"/{username}/{reponame}/issues",
-		"/{username}/{reponame}/{type:issues}",
-		"/{username}/{reponame}/pulls",
-		"/{username}/{reponame}/{type:pulls}",
-		"/{username}/{reponame}/{type:issues|pulls}", // for 1.23 only
-
-		// wiki
-		"/{username}/{reponame}/wiki/",
-
-		// activity
-		"/{username}/{reponame}/activity/",
 	}
 	for _, path := range expensivePaths {
 		if strings.HasPrefix(routePattern, path) {
-- 
2.45.2

