diff --git a/node_modules/fomantic-ui/tasks/build/css.js b/node_modules/fomantic-ui/tasks/build/css.js
index 3d2ae91..2d753f0 100644
--- a/node_modules/fomantic-ui/tasks/build/css.js
+++ b/node_modules/fomantic-ui/tasks/build/css.js
@@ -140,8 +140,11 @@ function buildCSS(src, type, config, opts, callback) {
   packCompressed.displayName = 'Packing compressed CSS';
 
   gulp.parallel(
-    gulp.series(buildUncompressed, packUncompressed),
-    gulp.series(buildCompressed, packCompressed)
+    gulp.series(
+      buildUncompressed,
+      gulp.parallel(packUncompressed, packCompressed)
+    ),
+    gulp.series(buildCompressed)
   )(callback);
 }
 
